src/server/req_select.c

```diff
@@ -494,6 +494,9 @@ select_job(job *pjob, struct select_list *psel, int dosubjobs, int dohistjobs)
 	else if ((dosubjobs != 2) &&
 		(pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob))
 		return 0;	/* don't bother to look at sub job */
+	else if ((dosubjobs == 2) && (pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob) &&
+		(pjob->ji_qs.ji_state != JOB_STATE_RUNNING)) /* select only running subjobs */
```

Comment:
Just checking, what about subjobs in states 'H', 'M', 'S', 'T', 'U' or 'W'? I'm not sure if subjobs can go into these states or not, but just in case, do we need to select subjobs in any of these states?

Reply:
Subjobs cannot go into some of the states like 'H', it can go into the suspended state but at that time the state is actually 'R' it is the sub-state that actually tells us whether it is suspended. For all other states, the scheduler does not really care.